bazaar/src/lib.rs

```diff
@@ -94,7 +94,7 @@ decl_module! {
 
         /// Allow a user to create a shop
         #[weight = 10_000]
-        pub fn upload_shop(origin, cid: CurrencyIdentifier, shop: ShopIdentifier) -> DispatchResult {
+        pub fn new_shop(origin, cid: CurrencyIdentifier, shop: ShopIdentifier) -> DispatchResult {
```

Comment:
one more thing, sorry:
I could upload a 1GB `Vec<u8>` here. You have to constrain the size of this vec to the size needed for IPFS URI
this applies to all dispatchables with dynamic size arguments!

Please also add a test where the dispatch maust fail with an oversize argument

The alternative would be to make the weight dependen t on the size of arguments, but that is of no use to you because Vec will always have the same length if used properly

Reply:
I have included a max length and the respective test.

Though not yet relevant, but it might be in the eventually:
IPFS might use different hashing algorithms in the future, which will result in longer byte lenghts. Maybe we should include tests on the app side (i.e. prepended multihash length coincides with hash digest) and change to weight depended blockchain inclusion?